cpp/src/arrow/testing/random.cc

```diff
@@ -721,13 +709,11 @@ std::shared_ptr<Array> RandomArrayGenerator::ArrayOf(const Field& field, int64_t
     for (const auto& length : *lengths) {                                            \
       if (length.has_value()) values_length += *length;                              \
     }                                                                                \
-    const auto force_empty_nulls =                                                   \
-        GetMetadata<bool>(field.metadata().get(), "force_empty_nulls", false);       \
     const auto values =                                                              \
         ArrayOf(*internal::checked_pointer_cast<ARRAY_TYPE::TypeClass>(field.type()) \
                      ->value_field(),                                                \
                 values_length);                                                      \
-    const auto offsets = OffsetsFromLengthsArray(lengths.get(), force_empty_nulls);  \
+    const auto offsets = OffsetsFromLengthsArray(lengths.get());                     \
```

Comment:
Hmm... instead of fixing `OffsetsFromLengthsArray` and ignoring `force_empty_nulls`, why not use `GenerateOffsets` or even `RandomArrayGenerator::List`?

Reply:
These don't allow limiting min_length and max_length for the individual lists, which at least some tests actually test (but without nulls), so I didn't want to break that.

Reply:
I see. Perhaps we can reconcile all these so as to have a single workhorse function that takes all desired parameters?

Reply:
That makes sense. I'll try to implement that